File: hw/pazmino/u3/HW30URIs/HW30URI/src/main/java/ec/edu/espe/hw30uri/model/Project.java
package ec.edu.espe.hw30uri.model;

import org.bson.Document;
import java.util.Date;

/**
 *
 * @author Brandon Pazmino
 */
public class Project {

    private String projectId;
    private String projectTitle;
    private String projectDescription;
    private String startDate;
    private String closingDate;
    private double startQuote;
    private String operationalStatus;
    private String quoteStatus;
    private boolean paid;
    private boolean invoiced;
    private boolean isPublic;

    public Project(String projectId, String projectTitle, String projectDescription, String startDate, String closingDate,
            double startQuote, String operationalStatus, String quoteStatus, boolean paid, boolean invoiced, boolean isPublic) {
        this.projectId = projectId;
        this.projectTitle = projectTitle;
        this.projectDescription = projectDescription;
        this.startDate = startDate;
        this.closingDate = closingDate;
        this.startQuote = startQuote;
        this.operationalStatus = operationalStatus;
        this.quoteStatus = quoteStatus;
        this.paid = paid;
        this.invoiced = invoiced;
        this.isPublic = isPublic;
    }

    public Document toDocument() {
        return new Document("projectId", projectId)
                .append("projectTitle", projectTitle)
                .append("projectDescription", projectDescription)
                .append("startDate", startDate)
                .append("closingDate", closingDate)
                .append("startquote", startQuote)
                .append("operationalStatus", operationalStatus)
                .append("quoteStatus", quoteStatus)
                .append("paid", paid)
                .append("invoiced", invoiced)
                .append("isPublic", isPublic);
    }
}

File: hw/pazmino/u3/HW30URIs/hw40URIPython/ec/espe/espe/hw40_uri_python/model/project.py
from pymongo import MongoClient

class Project:
    def __init__(self, project_id, project_title, project_description, start_date, closing_date,
                 start_quote, operational_status, quote_status, paid, invoiced, is_public):
        self.project_id = project_id
        self.project_title = project_title
        self.project_description = project_description
        self.start_date = start_date
        self.closing_date = closing_date
        self.start_quote = start_quote
        self.operational_status = operational_status
        self.quote_status = quote_status
        self.paid = paid
        self.invoiced = invoiced
        self.is_public = is_public

    def to_document(self):
        return {
            "projectId": self.project_id,
            "projectTitle": self.project_title,
            "projectDescription": self.project_description,
            "startDate": self.start_date,
            "closingDate": self.closing_date,
            "startquote": self.start_quote,
            "operationalStatus": self.operational_status,
            "quoteStatus": self.quote_status,
            "paid": self.paid,
            "invoiced": self.invoiced,
            "isPublic": self.is_public
        }
